ci/dev/watch.ts

```diff
@@ -1,139 +1,204 @@
-import * as cp from "child_process"
+import { spawn, fork, ChildProcess } from "child_process"
+import del from "del"
+import { promises as fs } from "fs"
 import * as path from "path"
-import { onLine } from "../../src/node/util"
-
-async function main(): Promise<void> {
-  try {
-    const watcher = new Watcher()
-    await watcher.watch()
-  } catch (error: any) {
-    console.error(error.message)
-    process.exit(1)
-  }
+import { CompilationStats, onLine, OnLineCallback, VSCodeCompileStatus } from "../../src/node/util"
+
+interface DevelopmentCompilers {
+  [key: string]: ChildProcess | undefined
+  vscode: ChildProcess
+  vscodeWebExtensions: ChildProcess
+  codeServer: ChildProcess
+  plugins: ChildProcess | undefined
 }
 
 class Watcher {
-  private readonly rootPath = path.resolve(__dirname, "../..")
-  private readonly vscodeSourcePath = path.join(this.rootPath, "vendor/modules/code-oss-dev")
+  private rootPath = path.resolve(process.cwd())
+  private readonly paths = {
+    /** Path to uncompiled VS Code source. */
+    vscodeDir: path.join(this.rootPath, "vendor", "modules", "code-oss-dev"),
+    compilationStatsFile: path.join(this.rootPath, "out", "watcher.json"),
+    pluginDir: process.env.PLUGIN_DIR,
+  }
+
+  //#region Web Server
 
-  private static log(message: string, skipNewline = false): void {
-    process.stdout.write(message)
-    if (!skipNewline) {
-      process.stdout.write("\n")
+  /** Development web server. */
+  private webServer: ChildProcess | undefined
+
+  private reloadWebServer = (): void => {
```

Comment:
❓ Basically this `reloadWebServer` kills the `webServer` if it exists, forks the process (not sure what `process.argv.slice(2)` is here? (might be nice to name it or add a comment?) and starts up a new child process, logging a message to let us know it started 

(and also adding a log for when the process exits)

Reply:
`process.argv.slice(2)` is a bit of a node convention. Since this array contains all the CLI arguments, the first two entries are usually the node runtime and the script name, e.g. `node entry.js`

Reply:
TIL! Thanks for the explainer!